src/interface.rs

```diff
@@ -418,7 +420,8 @@ impl Interface {
         }
 
         if settings.activate_connections {
-            connection.status = if connection.autoconnect {
+            connection.status = if connection.autoconnect && system_interfaces.contains(&self.name)
```

Comment:
I wonder if it would make sense to put a code block like:
```rust
    if settings.activate_connections {
        let system_interfaces = list_system_interfaces()?;
        for con in &mut connection_result.connections {
            if  con.autoconnect && system_interfaces.contains(&con.id) {
                con.status = Status::Up;
            }
        }
    }
```
to the end of `migrate::to_networkstate()`. Instead of changing the behavior of `to_connection()`

Reply:
Also, I wonder if it will work with "virtual" interfaces like `Dummy` or `bond`, when these interfaces are **not** created currently.

Reply:
I think you mean this but sure I think I like that better as well
```rust
    if settings.activate_connections {
        let system_interfaces = list_system_interfaces()?;
        for con in &mut connection_result.connections {
            if con.autoconnect && !system_interfaces.contains(&con.id) {
                con.status = agama_network::types::Status::Down;
            }
        }
    }
```

Reply:
ok, regarding my code snippet... it will not work. As the `con.id` might not be the interface name in case of wireless...

Reply:
> Also, I wonder if it will work with "virtual" interfaces like `Dummy` or `bond`, when these interfaces are **not** created currently.

Mhm very valid. I guess for dummy it should just be "blacklisted". For bonds I guess it should be "up" if the parent interface exists in the system? Which would also be in favor of the "post-process" approach in `migrate::to_networkstate()`